From 18adfdcff876ab0aee60dfb9b98a0f2e7cf30c81 Mon Sep 17 00:00:00 2001
From: Luis Ponce <luis.f.ponce.navarro@intel.com>
Date: Mon, 24 Feb 2020 22:27:12 +0000
Subject: [PATCH] sparksql could not start in jdk11

Exception org.datanucleus.exceptions.NucleusException: The java type
java.lang.Long (jdbc-type='', sql-type="") cant be mapped for this
datastore. No mapping is available.

SPARK-27538
---
 pom.xml          |  3 +++
 sql/hive/pom.xml | 15 +++++++++++++++
 2 files changed, 18 insertions(+)

diff --git a/pom.xml b/pom.xml
index a11ed536d3..6bb5686978 100644
--- a/pom.xml
+++ b/pom.xml
@@ -170,6 +170,9 @@
     <!-- org.apache.commons/commons-lang3/-->
     <commons-lang3.version>3.8.1</commons-lang3.version>
     <datanucleus-core.version>3.2.10</datanucleus-core.version>
+    <datanucleus-api-jdo.version>4.2.1</datanucleus-api-jdo.version>
+    <datanucleus-rdbms.version>4.1.7</datanucleus-rdbms.version>
+    <datanucleus-jdo.version>3.2.0-m3</datanucleus-jdo.version>
     <janino.version>3.0.9</janino.version>
     <jersey.version>2.22.2</jersey.version>
     <joda.version>2.9.3</joda.version>
diff --git a/sql/hive/pom.xml b/sql/hive/pom.xml
index b522bfeac1..374fcf8692 100644
--- a/sql/hive/pom.xml
+++ b/sql/hive/pom.xml
@@ -168,6 +168,21 @@
       <groupId>org.datanucleus</groupId>
       <artifactId>datanucleus-core</artifactId>
     </dependency>
+    <dependency>
+    <groupId>org.datanucleus</groupId>
+      <artifactId>datanucleus-api-jdo</artifactId>
+      <version>${datanucleus-api-jdo.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.datanucleus</groupId>
+      <artifactId>datanucleus-rdbms</artifactId>
+      <version>${datanucleus-rdbms.version}</version>
+    </dependency>
+    <dependency>
+    <groupId>org.datanucleus</groupId>
+      <artifactId>javax.jdo</artifactId>
+      <version>${datanucleus-jdo.version}</version>
+    </dependency>
     <dependency>
       <groupId>org.apache.thrift</groupId>
       <artifactId>libthrift</artifactId>
-- 
2.21.0
